src/credentials/CHIPCert.cpp

```diff
@@ -695,6 +698,144 @@ CHIP_ERROR ChipDN::GetCertFabricId(uint64_t & fabricId) const
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ChipDN::DecodeFromTLV(TLVReader & reader)
+{
+    CHIP_ERROR err;
+    TLVType outerContainer;
+    Tag tlvTag;
+    uint32_t tlvTagNum;
+    OID attrOID;
+    bool attrIsPrintableString;
+
+    static constexpr uint32_t kOID_AttributeIsPrintableString_Flag = 0x00000080;
+    static constexpr uint32_t kOID_AttributeType_Mask              = 0x0000007F;
+
+    // Enter the List TLV element that represents the DN in TLV format.
+    ReturnErrorOnFailure(reader.EnterContainer(outerContainer));
```

Comment:
Shouldn't something here be checking that we are in fact positioned on a list element?

Reply:
Added check.